cmd/lotus-chainwatch/syncer/sync.go

```diff
@@ -173,6 +182,10 @@ func (s *Syncer) Start(ctx context.Context) {
 						log.Errorw("failed to gather unsynced blocks", "error", err)
 					}
 
+					if err := s.storeCirculatingSupply(ctx, change.Val); err != nil {
+						log.Errorw("failed to store circulating supply", "error", err)
+					}
+
```

Comment:
A short term problem of this being attached to the syncer is that only tipsets which arrive while chainwatch is running will be processed here. Follow up work is planned to make sure back tipsets are processed properly.